default/AI/CombatRatingsAI.py

```diff
@@ -0,0 +1,364 @@
+    if stats_dict:
+        standard_fighter_stats = sorted([(v, k) for k, v in stats_dict.items()])[-1][1]
+        return ShipCombatStats(stats=standard_fighter_stats)
+    else:
+        return default_ship_stats()
+
+
+def default_ship_stats():
+    """ Return some ship stats to assume if no other intel is available.
+
+    :return: Some weak standard ship
+    :rtype: ShipCombatStats
+    """
+    attacks = (4.0, 1)
+    structure = 15
+    shields = 0
+    fighters = 0
+    launchrate = 0
+    fighter_damage = 0
+    return ShipCombatStats(stats=(attacks, structure, shields, fighters, launchrate, fighter_damage))
+
+
+class ShipCombatStats(object):
+    """Stores all relevant stats of a ship for combat strength evaluation."""
+    class BasicStats(object):
+        """Stores non-fighter-related stats."""
+        def __init__(self, stat_tuple=None, attacks=None, structure=1, shields=0):
+            """
+
+            :param stat_tuple:
+             :type stat_tuple: tuple|None
+            :param attacks:
+            :type attacks: dict|None
+            :param structure:
+            :type structure: int
+            :param shields:
+            :type shields: int
+            :return:
+            """
+            if stat_tuple and isinstance(stat_tuple, tuple):
+                attack_tuple, self.structure, self.shields = stat_tuple
+                self.attacks = tuple_to_dict(attack_tuple)
+            else:
+                self.structure = structure
+                self.shields = shields
+                self.attacks = attacks or {}
+
+        def get_stats(self, hashable=False):
+            """
+
+            :param hashable: if hashable, return tuple instead of attacks-dict
+            :return: attacks, structure, shields
+            """
+            if not hashable:
+                return self.attacks, self.structure, self.shields
+            else:
+                return dict_to_tuple(self.attacks), self.structure, self.shields
+
+        def __str__(self):
+            return str(self.get_stats())
+
+    class FighterStats(object):
+        """ Stores fighter-related stats """
+        def __init__(self, stat_tuple=None, capacity=0, launch_rate=0, damage=0):
+            if stat_tuple and isinstance(stat_tuple, tuple):
+                self.capacity, self.launch_rate, self.damage = stat_tuple
+            else:
+                self.capacity = capacity
+                self.launch_rate = launch_rate
+                self.damage = damage
+
+        def __str__(self):
+            return str(self.get_stats())
+
+        def get_stats(self):
+            """
+            :return: capacity, launch_rate, damage
+            """
+            return self.capacity, self.launch_rate, self.damage
+
+    def __init__(self, ship_id=-1, consider_refuel=False, stats=None):
+        self.__ship_id = ship_id
+        self._consider_refuel = consider_refuel
+        if stats:
```

Comment:
Two inner class, each have two way to create (magic happens inside single constructor)
You can make simple constructor and prepare arguments before call of it:

```python
b1, b2, b3, f1, f2, f3 = stats or [None] * 6
self._basic_stats = self.BasicStats(b1, b2. b3)
self._fighter_stats = self.FighterStats(f1, f2, f3)
```